tap_lever/streams/opportunities.py

```diff
@@ -10,8 +18,122 @@ class OpportunityStream(TimeRangeStream):
     TABLE = "opportunities"
     KEY_PROPERTIES = ["id"]
 
-    CACHE_RESULTS = True
-
     @property
     def path(self):
         return "/opportunities"
+
+    def sync(self, child_streams=None):
+        LOGGER.info('Syncing stream {} with {}'
+                    .format(self.catalog.tap_stream_id,
+                            self.__class__.__name__))
+
+        self.write_schema()
+
+        return self.sync_data(child_streams)
+
+    def sync_paginated(self, url, params=None, child_stream=None):
+        table = self.TABLE
+        _next = True
+        page = 1
+
+        all_resources = []
+        transformer = singer.Transformer()
+        while _next is not None:
+            result = self.client.make_request(url, self.API_METHOD, params=params)
+            _next = result.get('next')
+            data = self.get_stream_data(result['data'], transformer)
+
+            LOGGER.info('Starting Opportunity child stream syncs')
+            for opportunity in data:
+                opportunity_id = opportunity['id']
+
+                if child_stream.get('opportunity_applications'):
+                    OpportunityApplicationsStream(
```

Comment:
The class objects should be instantiated outside the `while` to save on object creation/unboxing/etc. time.

Reply:
Done